core/src/apps/cardano/sign_tx.py

```diff
@@ -146,18 +116,24 @@ def _process_outputs(self):
         change_coins = []
 
         for output in self.outputs:
-            if output.address_n:
-                address, _ = derive_address_and_node(self.keychain, output.address_n)
+            if output.address_parameters:
+                address = derive_address(
+                    self.keychain,
+                    output.address_parameters,
+                    self.protocol_magic,
+                    human_readable=False,
+                )
                 change_addresses.append(address)
-                change_derivation_paths.append(output.address_n)
+                change_derivation_paths.append(output.address_parameters.address_n)
                 change_coins.append(output.amount)
             else:
                 if output.address is None:
                     raise wire.ProcessError(
                         "Each output must have address or address_n field!"
                     )
-                if not is_safe_output_address(output.address):
-                    raise wire.ProcessError("Invalid output address!")
+                # todo: GK - this should be checked only with byron addresses
+                # if not is_safe_output_address(output.address):
+                #     raise wire.ProcessError("Invalid output address!")
 
                 outgoing_coins.append(output.amount)
```

Comment:
Do we still need to count coins? Or is it just some Byron relict?

Reply:
We still need to know, which coins belongs to which output. Maybe this whole class could now be rewritten and simplified. I've added to my todolist to consider rewriting it.